app/models/comment.rb

```diff
@@ -1,2 +1,8 @@
 class Comment < ApplicationRecord
+  belongs_to :user, foreign_key: 'user_id'
+  belongs_to :post, foreign_key: 'post_id'
+
+  def update_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```

Comment:
- [ Optional ] I think it would be nice if you make use of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comments_counter` method every time a new `Comment` has been saved to the database, then make the `update_comments_counter` method private. Please see the example below for a suggestion on how to achieve this.

## Example code
```ruby
class Comment < ApplicationRecord
  belongs_to :user, foreign_key: 'user_id'
  belongs_to :post, foreign_key: 'post_id'
  
  after_save :update_comments_counter
  
  private

  def update_comments_counter
    post.update(comments_counter: post.comments.count)
  end
end
```

Reply:
Thanks, @Felix45 for making all the suggestions to improve my logic, I truly appreciate it. I will make these changes and merge them. Have a great day 😄